File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque

input = sys.stdin.readline

que = deque()

for _ in range(int(input())):
    s = input()
    
    arr = s.split()
    
    if arr[0] == "push":
        que.append(arr[1])
    elif arr[0] == "front":
        print(-1 if not que else que[0])
    elif arr[0] == "back":
        print(-1 if not que else que[-1])
    elif arr[0] == "size":
        print(len(que))
    elif arr[0] == "empty":
        print(1 if len(que) == 0 else 0)
    elif arr[0] == "pop":
        print(-1 if not que else que.popleft())
    
    

File: 백준/Silver/10845. 큐/큐.js
const fs = require('fs')

const [n, ...inputs] = fs.readFileSync("/dev/stdin").toString().trim().split("\n")

const que = [];

for(let s of inputs){
    s = s.trim();
    const [a, b] = s.split(" ");
    
    if (a === "push"){
        que.push(b);
    } else if(a === "pop"){
        console.log(que.length ? que.shift() : -1);
    } else if(a === "size"){
        console.log(que.length)
    } else if (a === "empty"){
        console.log(que.length === 0 ? 1 : 0)
    } else if (a === "front"){
        console.log(que[0] ?? -1)
    } else if (a === "back"){
        console.log(que.at(-1) ?? -1)
    }
}